Improve column width handling in ExtensibleTable

fahrigedik opened the pull request:
Updated the table to use default widths when columnWidths are undefined and added canAutoResize binding. Added logic in the component to reset column widths if any are missing, ensuring consistent table rendering.

### Description

Resolves  https://github.com/volosoft/vs-internal/issues/7259  (write the related issue number if available)

### How to test it?

run dev app in volo repo

fahrigedik committed:
Improve column width handling in ExtensibleTable

Updated the table to use default widths when columnWidths are undefined and added canAutoResize binding. Added logic in the component to reset column widths if any are missing, ensuring consistent table rendering.

fahrigedik committed:
Revert "Improve column width handling in ExtensibleTable"

This reverts commit 53cfc6a81779c7e30b7bbfd1425940180dfa74a8.

sumeyyeKurtulus reviewed (commented):
Hello @fahrigedik thank you for coming up with such solution 🙌🏻 it works fine as it should be. However, we can switch to a signal based approach to make it better in terms of scalability and efficiency. Here is an example:

```ts
//...
  // Signal for actions column width
  private readonly _actionsColumnWidth = signal<number | undefined>(DEFAULT_ACTIONS_COLUMN_WIDTH);

  @Input() set actionsColumnWidth(width: number) {
    this._actionsColumnWidth.set(width ? Number(width) : undefined);
  }
  readonly columnWidths = computed(() => {
    const actionsColumn = this._actionsColumnWidth();
    const widths = [actionsColumn];
    this.propList.forEach(({ value: prop }) => {
      widths.push(prop.columnWidth);
    });
    return widths;
  });
  ```

fahrigedik committed:
Reapply "Improve column width handling in ExtensibleTable"

This reverts commit 3f41906037d0475ba9b484376e77e60004066802.

fahrigedik committed:
Refactor column width handling with signals and computed

Replaces manual column width management with Angular signals and computed properties for improved reactivity and code clarity in ExtensibleTableComponent. Removes the setColumnWidths method and updates the actionsColumnWidth input to use a signal.
